flocker/control/_model.py

```diff
@@ -301,9 +347,8 @@ def __invariant__(self):
         return (True, "")
 
     hostname = field(type=unicode, factory=unicode, mandatory=True)
-    applications = pset_field(Application)
-    manifestations = field(type=PMap, initial=pmap(), factory=pmap,
-                           mandatory=True)
+    applications = pset_field(Application, optional=True)
+    manifestations = pmap_field(unicode, Manifestation, optional=True)
```

Comment:
So there needs to be test and invariant needs to be fixed, but it needs to handle `None`, not absence of attribute.